File: shared/src/main/java/io/github/robertomike/hefesto/actions/wheres/BaseWhere.java
package io.github.robertomike.hefesto.actions.wheres;

import io.github.robertomike.hefesto.enums.WhereOperator;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

/**
 * The base class for all the where
 */
@Getter
@Setter
@ToString
@EqualsAndHashCode
public abstract class BaseWhere {
    /**
     * The operator that will be used
     */
    protected WhereOperator whereOperation = WhereOperator.AND;
}

File: shared/src/main/kotlin/io/github/robertomike/hefesto/actions/wheres/BaseWhere.kt
package io.github.robertomike.hefesto.actions.wheres

import io.github.robertomike.hefesto.enums.WhereOperator

/**
 * Base class for all WHERE conditions.
 * 
 * All WHERE clause types extend this class to support
 * logical operators (AND/OR) between conditions.
 */
abstract class BaseWhere {
    /**
     * The logical operator used to combine this condition with the previous one.
     * Defaults to AND.
     */
    open var whereOperation: WhereOperator = WhereOperator.AND
}
